framework/src/modules/token/token_module.ts

```diff
@@ -114,6 +117,12 @@ export class TokenModule extends BaseModule {
 				`Insufficient transaction fee. Minimum required fee is: ${minimumRequiredFee.toString()}`,
 			);
 		}
+
+		// Deduct transaction fee from sender balance
+		const senderAddress = transaction.senderID;
+		const sender = await stateStore.account.getOrDefault<TokenAccount>(senderAddress);
```

Comment:
sender must always exist so prefer `get`
```suggestion
		const sender = await stateStore.account.get<TokenAccount>(senderAddress);
```